Add introduction document of singleton python client sdk (#698)
vmware/singleton: content/en/docs/overview/singleton-sdk/python-client-introduction.md

---
title: "Python Client Introduction"
date: 2020-07-30T20:07:51+08:00
draft: false
weight: 10
---

# Introduction

> Several python files are provided as the python client to support accessing to Singleton service, making Singleton users to handle its globalization easily.

Reviewer: ```suggestion
Several python files are provided as the python client to support accessing Singleton service, making Singleton users to handle its globalization easily.
```

It needs a configuration file, like the [sample configuration file](https://github.com/vmware/singleton/blob/g11n-python-client/sgtn4python/sample/sample_online_offline.yml).

# APIs Available
* [Factory Class](#Factory-Class)
   * [I18n](#I18n)
* [Interface For Release](#Interface-For-Release)
   * [Release](#Release)
* [Interface For Configuration](#Interface-For-Configuration)
   * [Config](#Config)
* [Interface For Translation](#Interface-For-Translation)
   * [Translation](#Translation)

## Factory Class
### I18n

* It's the factory class that creates and initializes the configuration object and the release object.

```python
class I18n():
    def add_config_file(cls, config_file)
    def add_config(cls, base_path, config_data)
    def set_current_locale(cls, locale)
    def get_current_locale(cls)
    def get_release(cls, product, version)
```

### I18n / add_config_file

> * Load a configuration json or yaml file and initialize a correspondent release object.

Reviewer: TYPO: correspondent

Author: ```suggestion
* Load a configuration json or yaml file and initialize a corresponding release object.
```

| ------ | ------ | ------ |
| product | string | Product name |
| version | string | Version |

* Get release object.

| Return | Description |
| ------ | ------ |
| [Release](#Release) | Release object |

## Interface For Release
### Release
```python
class Release:
    def get_config(self)
    def get_translation(self)
```

### Release / get_config

* Get the configuration object owned by the release object.

| Return | Description |
| ------ | ------ |
| [Config](#Config) | The configuration object |

### Release / get_translation

* Get the translation object owned by the release object.

| Return | Description |
| ------ | ------ |
| [Translation](#Translation) | The translation object |

## Interface For Configuration
### Config
```python
class Config:
    def get_config_data(self)
    def get_info(self)
```
### Config / get_config_data

* Get configuration data.

| Return | Description |
| ------ | ------ |
| object | Configuration data |

### Config / get_info

* Get brief information of configuration.

| Return | Description |
| ------ | ------ |
| object | Brief information data |

## Interface For Translation
### Translation
```python
class Translation:

>     def get_string(self, component, key, **kwargs)

Reviewer: ```suggestion
    def get_string(self, component, key, **args)
```

Author: kwargs is a python style name that has fixed meaning.

    def get_locale_strings(self, locale)
    def get_locale_supported(self, locale)
```

### Translation / get_string

> * Get the trnalsation message of a component and a key.

Author: ```suggestion
* Get the translation message of a component and a key.
```

> * Get the trnalsation message of a component and a key.

Reviewer: ```suggestion
* Get the translation messages of a component and a key.
```

Author: Only one string is returned.

Reviewer: It has a typo
